hy/lex/lexer.py

```diff
@@ -14,6 +15,12 @@
 
 identifier = r'[^()\[\]{}\'"\s;]+'
 
+hashstring_paired_delims = (
+  # Unicode General Category "Pi" with the matching closing mark
+  ("«", "»"), ("‘", "’"), ("‛", "’"), ("“", "”"), ("‹", "›"), ("⸂", "⸃"), ("⸄", "⸅"), ("⸉", "⸊"), ("⸌", "⸍"), ("⸜", "⸝"), ("⸠", "⸡"),  # noqa
+  # BidiBrackets.txt
+  ("(", ")"), ("[", "]"), ("{", "}"), ("༺", "༻"), ("༼", "༽"), ("᚛", "᚜"), ("⁅", "⁆"), ("⁽", "⁾"), ("₍", "₎"), ("⌈", "⌉"), ("⌊", "⌋"), ("〈", "〉"), ("❨", "❩"), ("❪", "❫"), ("❬", "❭"), ("❮", "❯"), ("❰", "❱"), ("❲", "❳"), ("❴", "❵"), ("⟅", "⟆"), ("⟦", "⟧"), ("⟨", "⟩"), ("⟪", "⟫"), ("⟬", "⟭"), ("⟮", "⟯"), ("⦃", "⦄"), ("⦅", "⦆"), ("⦇", "⦈"), ("⦉", "⦊"), ("⦋", "⦌"), ("⦍", "⦐"), ("⦏", "⦎"), ("⦑", "⦒"), ("⦓", "⦔"), ("⦕", "⦖"), ("⦗", "⦘"), ("⧘", "⧙"), ("⧚", "⧛"), ("⧼", "⧽"), ("⸢", "⸣"), ("⸤", "⸥"), ("⸦", "⸧"), ("⸨", "⸩"), ("〈", "〉"), ("《", "》"), ("「", "」"), ("『", "』"), ("【", "】"), ("〔", "〕"), ("〖", "〗"), ("〘", "〙"), ("〚", "〛"), ("﹙", "﹚"), ("﹛", "﹜"), ("﹝", "﹞"), ("（", "）"), ("［", "］"), ("｛", "｝"), ("｟", "｠"), ("｢", "｣"))  # noqa
```

Comment:
There's no need to list all these out. https://docs.python.org/3.6/library/unicodedata.html